### ISSUE TITLE
Chart component Scatter type is crashing companion and apk. 

### ISSUE THREAD

**Issue by PeterMathijssen** [mit-cml/appinventor-sources#2735]
**Describe the bug**

Chart component Scatter type is crashing companion and apk. 

**Affects**

- [ ] Designer
- [ ] Blocks editor
- [X] Companion
- [X] Compiled apps
- [ ] Buildserver
- [ ] Debugging
- [X] Chart component

**Expected behavior**

Like all other Chart types the Scatter type shouldn't crash the companion of apk. 

**Steps to reproduce**

Use these blocks with the scatter type of the Chart component
![image](https://user-images.githubusercontent.com/32273650/192081383-18b25571-5c66-4cdc-8647-510b34096d9b.png)

https://community.appinventor.mit.edu/t/chart-component/66097

**Comment by ewpatton** [comment]
Confirmed that with the given blocks that this is an issue. I think the problem is that the scatter plot functionality expects that when adding an entry the X value is greater than or equal to the smallest X value. When randomly adding, this isn't guaranteed.

Here is a stack trace captured after implementing the above blocks:

```
09-24 02:17:43.698  4254  4254 E AndroidRuntime: FATAL EXCEPTION: main
09-24 02:17:43.698  4254  4254 E AndroidRuntime: Process: edu.mit.appinventor.aicompanion3, PID: 4254
09-24 02:17:43.698  4254  4254 E AndroidRuntime: java.lang.NegativeArraySizeException: -4
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at com.github.mikephil.charting.utils.Transformer.generateTransformedValuesScatter(Transformer.java:104)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at com.github.mikephil.charting.renderer.ScatterChartRenderer.drawValues(ScatterChartRenderer.java:116)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at com.github.mikephil.charting.charts.BarLineChartBase.onDraw(BarLineChartBase.java:278)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.view.View.draw(View.java:19123)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:18073)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4198)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4178)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:18032)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4198)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4178)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:18032)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4198)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4178)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:18032)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4198)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4178)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:18032)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4198)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4178)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:18032)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4198)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4178)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:18032)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4198)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4178)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:18032)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4198)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4178)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:18032)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4198)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4178)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:18032)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:4198)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:4178)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:18032)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.view.ThreadedRenderer.updateViewTreeDisplayList(ThreadedRenderer.java:643)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.view.ThreadedRenderer.updateRootDisplayList(ThreadedRenderer.java:649)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.view.ThreadedRenderer.draw(ThreadedRenderer.java:757)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.view.ViewRootImpl.draw(ViewRootImpl.java:2980)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.view.ViewRootImpl.performDraw(ViewRootImpl.java:2794)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2347)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1386)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:6733)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:911)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.view.Choreographer.doCallbacks(Choreographer.java:723)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.view.Choreographer.doFrame(Choreographer.java:658)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:897)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:789)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:98)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:164)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:6541)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
09-24 02:17:43.698  4254  4254 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
09-24 02:17:43.703  1672  3221 W ActivityManager:   Force finishing activity edu.mit.appinventor.aicompanion3/.Screen1
```

**Comment by ewpatton** [comment]
Fixed in companion 2.66
